src/main/java/ch/njol/skript/expressions/ExprExplosionCause.java

```diff
@@ -0,0 +1,75 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.bukkit.event.block.BlockExplodeEvent;
+import org.bukkit.event.entity.EntityExplodeEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Explosion Cause")
+@Description("Get explosion cause in 'on explode' event")
+@Examples({"on explode:",
+	"\texplosion type = \"block\" # Either \"block\" or \"entity\""})
+@Events("explode")
+@Since("2.5")
+public class ExprExplosionCause extends SimpleExpression<String> {
```

Comment:
I don't think this expression should be added

Reply:
Was originally suggested by Frank, and I think this is needed in case someone wanted to do multiple codes using one general `on explode` method, so this is useful is here

Reply:
The only thing I don't like about it is that it returns a text

Reply:
So what Frank meant by https://github.com/SkriptLang/Skript/pull/3990#pullrequestreview-657573800, was that the expression is supposed to return an Entity, or a Block, not the string `"entity"` or `"block"`. This way, a user could use `if explosion cause is an entity:` to check

Reply:
That's the thing I was trying to do but I didn't know how to add a type in skript

Reply:
You don't have to create a new type for that, it'll be a `Expression<Object>` that returns either an Entity or a Block, depending on the event.